src/Microsoft.Diagnostics.Runtime/src/Common/ClrGenericParameter.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    public readonly struct ClrGenericParameter
+    {
+        public int Index { get; }
+
+        public GenericParameterAttributes Attributes { get; }
+
+        public string Name { get; }
+
+        internal ClrGenericParameter(int index, GenericParameterAttributes attributes, string name)
```

Comment:
We should add the metadata token for this generic parameter in case the user wants to grab more data out of IMetaDataReader in the future.